src/Features/CSharp/Portable/CodeFixes/AddExplicitCast/AddExplicitCastCodeFixProvider.cs

```diff
@@ -0,0 +1,399 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Internal.Log;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Simplification;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+#nullable enable
+namespace Microsoft.CodeAnalysis.CSharp.CodeFixes.AddExplicitCast
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = PredefinedCodeFixProviderNames.AddExplicitCast), Shared]
+    [ExtensionOrder(After = PredefinedCodeFixProviderNames.ChangeToYield, Before = PredefinedCodeFixProviderNames.GenerateConstructor)]
+    internal partial class AddExplicitCastCodeFixProvider : SyntaxEditorBasedCodeFixProvider
+    {
+        /// <summary>
+        /// CS0266: Cannot implicitly convert from type 'x' to 'y'. An explicit conversion exists (are you missing a cast?)
+        /// </summary>
+        private const string CS0266 = nameof(CS0266);
+
+        /// <summary>
+        /// CS1503: Argument 1: cannot convert from 'double' to 'int'
+        /// </summary>
+        private const string CS1503 = nameof(CS1503);
+        private const int MaximumConversionOptions = 3;
+
+        [ImportingConstructor]
+        public AddExplicitCastCodeFixProvider()
+        {
+        }
+
+        public sealed override ImmutableArray<string> FixableDiagnosticIds
+        {
+            get => ImmutableArray.Create(CS0266, CS1503);
+        }
+
+        internal sealed override CodeFixCategory CodeFixCategory => CodeFixCategory.CodeStyle;
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var document = context.Document;
+            var cancellationToken = context.CancellationToken;
+            var diagnostic = context.Diagnostics.First();
+
+            var root = await document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+            if (root != null && syntaxFacts != null && semanticModel != null &&
```

Comment:
I don't believe you need to check for null values for any of these three variables.

Reply:
No, I don't need to. But since I add `#nullable enable` to the top of this file, the IDE gonna mark these variables as code smells because they are possible to be null. Or I can only put the  `#nullable enable` above the functions that need to be annotated?

Reply:
`GetRequiredXXX` overloads that I suggested above ensure that return values are always non-null, so you should be able to remove these checks.